Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -82,6 +82,11 @@ public struct KlaviyoSDK {
     public func resetProfile() {
         dispatchOnMainThread(action: .resetProfile)
     }
+    
+    public func resetBadgeCount(defaults: String) {
+        dispatchOnMainThread(action: .resetBadgeCount(defaults))
+        UIApplication.shared.applicationIconBadgeNumber = 0
```

Comment:
two comments:
1) I think this needs to be handled within the `.resetBadgeCount(defaults)` case in the `KlaviyoReducer.reduce` method
2) setting the badge number using `applicationIconBadgeNumber` has [been deprecated in iOS 17](https://developer.apple.com/documentation/uikit/uiapplication/1622918-applicationiconbadgenumber). We will need to handle the pre-iOS 16 case though, so we should use `if #available` to switch between the old API and the new one. This should work:
```swift
if #available(iOS 16.0, *) {
    UNUserNotificationCenter.current().setBadgeCount(0)
} else {
    UIApplication.shared.applicationIconBadgeNumber = 0
}
```
note that you'll need to add both `import UIKit` and `import UserNotifications` for this code to work